cassandra/src/test/java/org/hibernate/ogm/datastore/cassandra/test/mapping/ElementCollectionListWithIndexTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.cassandra.test.mapping;
+
+import static org.hibernate.ogm.datastore.cassandra.utils.CassandraTestHelper.rowAssertion;
+
+import org.hibernate.Transaction;
+import org.hibernate.ogm.OgmSession;
+import org.hibernate.ogm.backendtck.associations.collection.types.GrandChild;
+import org.hibernate.ogm.backendtck.associations.collection.types.GrandMother;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Nicola Ferraro
+ */
+public class ElementCollectionListWithIndexTest extends OgmTestCase {
+
+	private GrandMother granny;
+
+	@Before
+	public void prepareDb() throws Exception {
+		OgmSession session = openSession();
+		Transaction transaction = session.beginTransaction();
+
+		GrandChild luke = new GrandChild();
+		luke.setName( "Luke" );
+
+		GrandChild leia = new GrandChild();
+		leia.setName( "Leia" );
+
+		granny = new GrandMother();
+		granny.getGrandChildren().add( luke );
+		granny.getGrandChildren().add( leia );
+
+		session.persist( granny );
+		transaction.commit();
+		session.close();
+	}
+
+	@Test
+	public void testMapping() throws Exception {
+		OgmSession session = openSession();
+		Transaction transaction = session.beginTransaction();
+
+		rowAssertion( session.getSessionFactory(), "GrandMother" )
+				.id( "id", granny.getId() )
+				.assertNoOtherColumnPresent()
+				.execute();
+
+		rowAssertion( session.getSessionFactory(), "GrandMother_grandChildren" )
```

Comment:
I like that approach, it might even allow to implement it identically for other dialects.

I'd only rename `id()` into `keyColumn`, because that's what it is: one id/key comprising of several columns, not several ids.